### ISSUE TITLE
Plugin error from fragment-insertion:  Command '['run-sepp.sh' returned non-zero exit status 1

### ISSUE THREAD

**Issue by heronne** [qiime2/q2-fragment-insertion#49]
**Bug Description**
Hello I've been trying to use q2-fragment insertion in order to use PICRUSt2, following the instructions from the original source, unfortunately I got an error from this plug in, in some forum I saw the same error, and followed the instructions using this command:
first I tried it with the files of my interest but then I tried the files provided in the tutorial
qiime fragment-insertion sepp --i-representative-sequences mammal_seqs.qza --p-threads 12 --i-reference-alignment reference.fna.qza  --i-reference-phylogeny reference.tre.qza --output-dir pruebapicrust2tutorial --p-debug --verbose 2> err.txt > out.txt
there was no follow up on the error.

**References**
in order to view a more detailed information here are the files
[err.txt](https://github.com/biocore/q2-fragment-insertion/files/2506701/err.txt)
[out.txt](https://github.com/biocore/q2-fragment-insertion/files/2506704/out.txt)

**Comments**
I'm using an hp with the following hardware:
AMD® A12-9720p radeon r7, 12 compute cores 4c+8g × 4 

I though it may be a problem with the installation, so I removed qiime2 and reinstalled. I updated Anaconda and conda to the latest version.
Thank you

**Comment by sjanssen2** [comment]
Hi,
I was able to reproduce the expected results with your command for the picrust data. Note that the run used ~42 GB memory. Might that be an issue on your machine? The failure occurs on the memory consuming pplacer step.
Could you also send me the result of the following command, executed in your qiime environment:
```conda list```?
Thanks
Stefan

**Comment by heronne** [comment]
Yes of course, Thank you, I see, the memory might be a problem, not sure but sounds like it is. The output for `conda list` is:
`# packages in environment at /home/hnavar/anaconda3/envs/qiime2:
#
# Name                    Version                   Build  Channel
appdirs                   1.4.3                      py_1    conda-forge
asn1crypto                0.24.0                   py35_3    conda-forge
atomicwrites              1.2.1                     <pip>
attrs                     18.1.0                     py_1    conda-forge
automat                   0.7.0                      py_1    conda-forge
backcall                  0.1.0                      py_0    conda-forge
bibtexparser              1.0.1                      py_1    conda-forge
bioconductor-biobase      2.38.0           r341h219a75b_1    bioconda
bioconductor-biocgenerics 0.24.0                   r341_1    bioconda
bioconductor-biocparallel 1.12.0           r341h219a75b_1    bioconda
bioconductor-biostrings   2.46.0                 r3.4.1_0    bioconda
bioconductor-dada2        1.6.0                  r3.4.1_0    bioconda
bioconductor-delayedarray 0.4.1                  r3.4.1_0    bioconda
bioconductor-genomeinfodb 1.14.0                 r3.4.1_0    bioconda
bioconductor-genomeinfodbdata 1.0.0                  r3.4.1_1    bioconda
bioconductor-genomicalignments 1.14.1                 r3.4.1_0    bioconda
bioconductor-genomicranges 1.30.3                 r3.4.1_0    bioconda
bioconductor-iranges      2.12.0                 r3.4.1_0    bioconda
bioconductor-rsamtools    1.30.0                 r3.4.1_0    bioconda
bioconductor-s4vectors    0.16.0                 r3.4.1_0    bioconda
bioconductor-shortread    1.36.0                 r3.4.1_0    bioconda
bioconductor-summarizedexperiment 1.8.0                  r3.4.1_0    bioconda
bioconductor-xvector      0.18.0                 r3.4.1_0    bioconda
bioconductor-zlibbioc     1.24.0                 r3.4.1_0    bioconda
biom-format               2.1.6                    py35_8    qiime2
blas                      1.0                         mkl  
blast                     2.6.0               boost1.64_2    bioconda
bleach                    2.1.4                      py_1    conda-forge
bokeh                     0.13.0                   py35_0    conda-forge
boost                     1.64.0                   py35_4    conda-forge
boost-cpp                 1.64.0                        1    conda-forge
bz2file                   0.98                       py_0    conda-forge
bzip2                     1.0.6                h470a237_2    conda-forge
ca-certificates           2018.8.24            ha4d7672_0    conda-forge
cachecontrol              0.12.5                     py_0    conda-forge
cairo                     1.14.10                       0    conda-forge
certifi                   2018.8.24             py35_1001    conda-forge
cffi                      1.11.5           py35h5e8e0c9_1    conda-forge
chardet                   3.0.4                    py35_3    conda-forge
click                     6.7                        py_1    conda-forge
constantly                15.1.0                     py_0    conda-forge
coverage                  4.5.1                     <pip>
cryptography              2.3.1            py35hdffb7b8_0    conda-forge
cryptography-vectors      2.3.1                    py35_0    conda-forge
curl                      7.61.0               h93b3f91_2    conda-forge
cutadapt                  1.17                     py35_0    bioconda
cycler                    0.10.0                     py_1    conda-forge
cython                    0.28.5           py35hfc679d8_0    conda-forge
dbus                      1.13.0               h3a4f0e9_0    conda-forge
deblur                    1.0.4                    py35_2    bioconda
decorator                 4.3.0                      py_0    conda-forge
emperor                   1.0.0beta17              py35_1    conda-forge
entrypoints               0.2.3                    py35_2    conda-forge
expat                     2.2.5                hfc679d8_2    conda-forge
fastcluster               1.1.25           py35hf8a1672_0    conda-forge
fasttree                  2.1.10                        0    bioconda
fontconfig                2.12.6                        0    conda-forge
fragment-insertion        4.3.5                    py35_0    biocore
freetype                  2.8.1                hfa320df_1    conda-forge
future                    0.16.0                   py35_2    conda-forge
gettext                   0.19.8.1             h5e8e0c9_1    conda-forge
glib                      2.55.0               h464dc38_2    conda-forge
glpk                      4.65                 h16a7912_1    conda-forge
gmp                       6.1.2                hfc679d8_0    conda-forge
gneiss                    0.4.4            py27h1cf38a6_1    biocore
graphite2                 1.3.12               hfc679d8_1    conda-forge
gsl                       2.1                           2    conda-forge
gst-plugins-base          1.12.5               hde13a9d_0    conda-forge
gstreamer                 1.12.5               h61a6719_0    conda-forge
h5py                      2.7.0               np112py35_0    conda-forge
harfbuzz                  1.7.6                         0    conda-forge
hdf5                      1.8.17                       11    conda-forge
hdmedians                 0.13             py35h7eb728f_1    conda-forge
hmmer                     3.2.1                hfc679d8_0    bioconda
html5lib                  1.0.1                      py_0    conda-forge
hyperlink                 17.3.1                     py_0    conda-forge
icu                       58.2                 hfc679d8_0    conda-forge
idna                      2.7                      py35_2    conda-forge
ijson                     2.3                      py35_0    qiime2
incremental               17.5.0                     py_0    conda-forge
intel-openmp              2018.0.3                      0  
ipykernel                 4.9.0                    py35_0    conda-forge
ipython                   6.5.0                    py35_0    conda-forge
ipython_genutils          0.2.0                      py_1    conda-forge
ipywidgets                7.4.0                      py_0    conda-forge
iqtree                    1.6.7.1              he941832_0    bioconda
java-jdk                  8.0.92                        1    bioconda
jedi                      0.12.1                   py35_0    conda-forge
jinja2                    2.10                       py_1    conda-forge
joblib                    0.12.5                     py_0    conda-forge
jpeg                      9c                   h470a237_1    conda-forge
jsonschema                2.6.0                    py35_2    conda-forge
jupyter_client            5.2.3                      py_1    conda-forge
jupyter_core              4.4.0                      py_0    conda-forge
kiwisolver                1.0.1            py35h2d50403_2    conda-forge
krb5                      1.14.6                        0    conda-forge
libffi                    3.2.1                hfc679d8_5    conda-forge
libgcc                    7.2.0                h69d50b8_2    conda-forge
libgcc-ng                 7.2.0                hdf63c60_3    conda-forge
libgfortran               3.0.0                         1    conda-forge
libgfortran-ng            7.2.0                hdf63c60_3    conda-forge
libiconv                  1.15                 h470a237_3    conda-forge
libpng                    1.6.34               ha92aebf_2    conda-forge
libsodium                 1.0.16               h470a237_1    conda-forge
libssh2                   1.8.0                h5b517e9_2    conda-forge
libstdcxx-ng              7.2.0                hdf63c60_3    conda-forge
libtiff                   4.0.9                he6b73bb_2    conda-forge
libxcb                    1.13                 h470a237_2    conda-forge
libxml2                   2.9.8                h422b904_5    conda-forge
lockfile                  0.12.2                     py_1    conda-forge
mafft                     7.310                         1    bioconda
markupsafe                1.0              py35h470a237_1    conda-forge
matplotlib                2.2.2                    py35_1    conda-forge
mistune                   0.8.3            py35h470a237_2    conda-forge
mkl                       2018.0.3                      1  
more-itertools            4.3.0                     <pip>
msgpack-python            0.5.6            py35h2d50403_3    conda-forge
natsort                   5.3.3                      py_1    conda-forge
nbconvert                 5.3.1                      py_1    conda-forge
nbformat                  4.4.0                      py_1    conda-forge
ncurses                   5.9                          10    conda-forge
nodejs                    8.11.3               hf484d3e_0  
nose                      1.3.7                    py35_2    conda-forge
notebook                  5.6.0                    py35_1    conda-forge
numpy                     1.12.1           py35hca0bb5e_1  
openssl                   1.0.2p               h470a237_1    conda-forge
packaging                 17.1                       py_0    conda-forge
pandas                    0.22.0                   py35_1    conda-forge
pandoc                    2.2.2                         1    conda-forge
pandocfilters             1.4.2                      py_1    conda-forge
pango                     1.40.14                       0    conda-forge
parso                     0.3.1                      py_0    conda-forge
pathlib2                  2.3.2                     <pip>
patsy                     0.5.0                      py_1    conda-forge
pcre                      8.39                          0    conda-forge
pexpect                   4.6.0                    py35_0    conda-forge
pickleshare               0.7.4                    py35_0    conda-forge
PICRUSt2                  2.0.3b0                   <pip>
pigz                      2.3.4                         0    conda-forge
pip                       18.0                  py35_1001    conda-forge
pixman                    0.34.0               h470a237_3    conda-forge
pluggy                    0.8.0                     <pip>
prometheus_client         0.3.0                      py_0    conda-forge
prompt_toolkit            1.0.15                     py_1    conda-forge
psutil                    5.4.7            py35h470a237_1    conda-forge
pthread-stubs             0.4                  h470a237_1    conda-forge
ptyprocess                0.6.0                 py35_1000    conda-forge
py                        1.7.0                     <pip>
pyasn1                    0.4.4                      py_1    conda-forge
pyasn1-modules            0.2.1                      py_0    conda-forge
pycparser                 2.18                       py_1    conda-forge
pygments                  2.2.0                      py_1    conda-forge
pyhamcrest                1.9.0                      py_2    conda-forge
pyopenssl                 18.0.0                   py35_0    conda-forge
pyparsing                 2.2.0                      py_1    conda-forge
pyqt                      5.6.0            py35h8210e8a_7    conda-forge
pysocks                   1.6.8                    py35_2    conda-forge
pytest                    3.9.2                     <pip>
pytest-cov                2.6.0                     <pip>
python                    3.5.5                         1    conda-forge
python-dateutil           2.7.3                      py_0    conda-forge
pytz                      2018.5                     py_0    conda-forge
pyyaml                    3.13             py35h470a237_1    conda-forge
pyzmq                     17.1.2           py35hae99301_0    conda-forge
q2-alignment              2018.8.0                 py35_0    qiime2/label/r2018.8
q2-composition            2018.8.0                 py35_0    qiime2/label/r2018.8
q2-cutadapt               2018.8.0                 py35_0    qiime2/label/r2018.8
q2-dada2                  2018.8.0                 py35_0    qiime2/label/r2018.8
q2-deblur                 2018.8.0                 py35_0    qiime2/label/r2018.8
q2-demux                  2018.8.0                 py35_0    qiime2/label/r2018.8
q2-diversity              2018.8.0                 py35_0    qiime2/label/r2018.8
q2-emperor                2018.8.0                 py35_0    qiime2/label/r2018.8
q2-feature-classifier     2018.8.0                 py35_0    qiime2/label/r2018.8
q2-feature-table          2018.8.0                 py35_0    qiime2/label/r2018.8
q2-fragment-insertion     2018.6.17                py35_1    biocore
q2-gneiss                 2018.8.0                 py35_0    qiime2/label/r2018.8
q2-longitudinal           2018.8.0                 py35_0    qiime2/label/r2018.8
q2-metadata               2018.8.0                 py35_0    qiime2/label/r2018.8
q2-phylogeny              2018.8.0                 py35_0    qiime2/label/r2018.8
q2-picrust2               0.0.1                     <pip>
q2-quality-control        2018.8.0                 py35_0    qiime2/label/r2018.8
q2-quality-filter         2018.8.0                 py35_0    qiime2/label/r2018.8
q2-sample-classifier      2018.8.0                 py35_0    qiime2/label/r2018.8
q2-taxa                   2018.8.0                 py35_0    qiime2/label/r2018.8
q2-types                  2018.8.0                 py35_0    qiime2/label/r2018.8
q2-vsearch                2018.8.0                 py35_0    qiime2/label/r2018.8
q2cli                     2018.8.0                 py35_0    qiime2/label/r2018.8
q2templates               2018.8.0                 py35_0    qiime2/label/r2018.8
qiime2                    2018.8.0                 py35_0    qiime2/label/r2018.8
qt                        5.6.2                         7    conda-forge
r-assertthat              0.2.0            r341h6115d3f_1    conda-forge
r-base                    3.4.1                         4    conda-forge
r-bh                      1.66.0_1                 r341_1    conda-forge
r-bitops                  1.0_6            r341hc070d10_2    conda-forge
r-castor                  1.3.4            r341hfc679d8_0    conda-forge
r-cli                     1.0.0            r341h6115d3f_1    conda-forge
r-colorspace              1.3_2            r341hc070d10_2    conda-forge
r-crayon                  1.3.4            r341h6115d3f_1    conda-forge
r-data.table              1.11.4           r341hc070d10_2    conda-forge
r-dichromat               2.0_0                  r3.4.1_0    conda-forge
r-digest                  0.6.16           r341hc070d10_1    conda-forge
r-formatr                 1.5              r341h6115d3f_1    conda-forge
r-futile.logger           1.4.3            r341h6115d3f_1    conda-forge
r-futile.options          1.0.0                  r3.4.1_0    conda-forge
r-ggplot2                 3.0.0            r341h6115d3f_1    conda-forge
r-glue                    1.3.0            r341h470a237_2    conda-forge
r-gtable                  0.2.0            r341h6115d3f_1    conda-forge
r-hwriter                 1.3.2            r341h6115d3f_1    conda-forge
r-labeling                0.3              r341h6115d3f_1    conda-forge
r-lambda.r                1.2.3            r341h6115d3f_0    conda-forge
r-lattice                 0.20_35          r341hc070d10_0    conda-forge
r-latticeextra            0.6_28           r341h6115d3f_1    conda-forge
r-lazyeval                0.2.1            r341hc070d10_2    conda-forge
r-magrittr                1.5              r341h6115d3f_1    conda-forge
r-mass                    7.3_50           r341hc070d10_2    conda-forge
r-matrix                  1.2_14           r341hc070d10_2    conda-forge
r-matrixstats             0.53.1           r341h470a237_1    conda-forge
r-mgcv                    1.8_24           r341hc070d10_2    conda-forge
r-munsell                 0.5.0            r341h6115d3f_1    conda-forge
r-naturalsort             0.1.3                  r3.4.1_0    conda-forge
r-nlme                    3.1_131          r341ha44fe06_1    conda-forge
r-pillar                  1.2.2            r341h6115d3f_1    conda-forge
r-plyr                    1.8.4            r341h9d2a408_2    conda-forge
r-r6                      2.2.2            r341h6115d3f_1    conda-forge
r-rcolorbrewer            1.1_2            r341h6115d3f_1    conda-forge
r-rcpp                    0.12.17          r341h9d2a408_2    conda-forge
r-rcppparallel            4.4.0            r341hfc679d8_1    conda-forge
r-rcurl                   1.95_4.11        r341h86b9717_2    conda-forge
r-reshape2                1.4.3            r341h9d2a408_2    conda-forge
r-rlang                   0.2.1            r341h470a237_2    conda-forge
r-scales                  1.0.0            r341h9d2a408_1    conda-forge
r-snow                    0.4_2                  r3.4.1_0    conda-forge
r-stringi                 1.2.3            r341hfc679d8_1    conda-forge
r-stringr                 1.3.1            r341h6115d3f_1    conda-forge
r-tibble                  1.4.2            r341hc070d10_2    conda-forge
r-utf8                    1.1.4            r341hc070d10_0    conda-forge
r-viridislite             0.3.0            r341h6115d3f_1    conda-forge
r-withr                   2.1.1                  r3.4.1_0    conda-forge
raxml                     8.2.10               h470a237_1    bioconda
readline                  7.0                           0    conda-forge
requests                  2.19.1                   py35_1    conda-forge
scikit-bio                0.5.4                    py35_0    conda-forge
scikit-learn              0.19.1           py35hbf1f462_0  
scipy                     0.19.1           py35ha8f041b_3  
seaborn                   0.9.0                      py_0    conda-forge
send2trash                1.5.0                      py_0    conda-forge
service_identity          17.0.0                     py_0    conda-forge
setuptools                40.2.0                   py35_0    conda-forge
simplegeneric             0.8.1                      py_1    conda-forge
sip                       4.18.1           py35hfc679d8_0    conda-forge
six                       1.11.0                   py35_1    conda-forge
sortmerna                 2.0                  he941832_3    bioconda
sqlite                    3.20.1                        2    conda-forge
statsmodels               0.9.0                    py35_0    conda-forge
terminado                 0.8.1                    py35_1    conda-forge
testpath                  0.3.1                    py35_1    conda-forge
tk                        8.6.8                ha92aebf_0    conda-forge
tornado                   5.1.1            py35h470a237_0    conda-forge
traitlets                 4.3.2                    py35_0    conda-forge
twisted                   18.7.0           py35h470a237_1    conda-forge
tzlocal                   1.3                      py35_0    qiime2
unifrac                   0.9.2            py35hae32d81_1    biocore
urllib3                   1.23                     py35_1    conda-forge
vsearch                   2.7.0                         1    bioconda
wcwidth                   0.1.7                      py_1    conda-forge
webencodings              0.5.1                      py_1    conda-forge
wget                      1.19.5               h1ad7b7a_0  
wheel                     0.31.1                py35_1001    conda-forge
widgetsnbextension        3.4.0                    py35_1    conda-forge
xopen                     0.3.2                      py_1    bioconda
xorg-libxau               1.0.8                h470a237_6    conda-forge
xorg-libxdmcp             1.1.2                h470a237_7    conda-forge
xz                        5.2.4                h470a237_1    conda-forge
yaml                      0.1.7                h470a237_1    conda-forge
zeromq                    4.2.5                hfc679d8_6    conda-forge
zlib                      1.2.11               h470a237_3    conda-forge
zope.interface            4.5.0            py35h470a237_1    conda-forge`

**Comment by matrs** [comment]
I just had the same problem in a machine without enough memory (`qiime 2019.7`). The error message from `run-sepp.sh` is  a bit cryptic given that It doesn't say anything about hardware resources. Maybe an error with that specific information is hard to generate, but it would be great if some numbers about required resources are written into the official `qiime` tutorials. The same happened to me with the naive bayes classifier and I think a short phrase about max memory consumption would be very helpful for users.

**Comment by profgiuseppe** [comment]
Given that I had to solve this very problem for a client, I'm going to share here my findings, hoping they may help. I'm going to share it also on https://forum.qiime2.org/t/how-to-solve-sepp-error/13844/10 if I find time.

TL;DR: the pplacer step of fragment insertion uses vsyscalls, which are now disabled by default in most Linux kernels.

Installing a newver version of pplacer from bioconda does solves this issue, but raises another one in the subsequent step (the json merging java program) because probably there was a change in the structure of pplacer json output.

Then, the only solution we found was to enable vsyscall using vsyscall=emulate in the kernel boot string. In that way, the standard qiime2 fragment insertion installation works without trouble (at leat on the mouse example data).

This is another example of the need of maintenance and follow-up in code development for research software.

**Comment by sjanssen2** [comment]
Hi @profgiuseppe , thanks for your report!
As far as I remember, pplacer is no longer maintained, thus I wonder if the latest version (v1.1.alpha19-0-g807f6f3) comes without the vsyscall use?

Unfortunately, compilation of pplacer isn't trivial - or maybe it is my insufficient compilation skills. However, the latest version in bioconda is lacking an OSx package :-/

Thus, if the latest version would work, we would need to a) compile also for OSx and b) double check if results remain the same. Any help on a) is much appreachiated!

**Comment by antgonza** [comment]
I'm a bit confused on how to reproduce this error or exactly which systems are affected by it. Any thoughts?

BTW I basically just conda installed the latest qiime2 in our systems and then ran:
```bash
$ wget -O "sepp-refs-gg-13-8.qza" "https://data.qiime2.org/2020.2/common/sepp-refs-gg-13-8.qza"
$ wget https://docs.qiime2.org/2020.2/data/tutorials/pd-mice/dada2_rep_set.qza
$ qiime fragment-insertion sepp --i-representative-sequences ./dada2_rep_set.qza --i-reference-database sepp-refs-gg-13-8.qza --o-tree ./tree.qza --o-placements ./tree_placements.qza --p-threads 1
/home/antoniog/miniconda3/envs/qiime2/lib/python3.6/site-packages/ijson/backends/__init__.py:22: UserWarning: Cannot determine yajl version, assuming <1.0.12
  warnings.warn('Cannot determine yajl version, assuming <1.0.12')

Saved Phylogeny[Rooted] to: ./tree.qza
Saved Placements to: ./tree_placements.qza
```

In case it helps:
```
$ qiime info
System versions
Python version: 3.6.7
QIIME 2 release: 2020.2
QIIME 2 version: 2020.2.0
q2cli version: 2020.2.0

Installed plugins
alignment: 2020.2.0
composition: 2020.2.0
cutadapt: 2020.2.0
dada2: 2020.2.0
deblur: 2020.2.0
demux: 2020.2.0
diversity: 2020.2.0
emperor: 2020.2.0
empress: 0.1.dev0
feature-classifier: 2020.2.0
feature-table: 2020.2.0
fragment-insertion: 2020.2.0
gneiss: 2020.2.0
longitudinal: 2020.2.0
metadata: 2020.2.0
phylogeny: 2020.2.0
quality-control: 2020.2.0
quality-filter: 2020.2.0
sample-classifier: 2020.2.0
taxa: 2020.2.0
types: 2020.2.0
vsearch: 2020.2.0

Application config directory
/home/antoniog/miniconda3/envs/qiime2/var/q2cli

Getting help
To get help with QIIME 2, visit https://qiime2.org
```

**Comment by profgiuseppe** [comment]
> Hi @profgiuseppe , thanks for your report!
> As far as I remember, pplacer is no longer maintained, thus I wonder if the latest version (v1.1.alpha19-0-g807f6f3) comes without the vsyscall use?

@sjanssen2 , alpha19 does not use the vsyscall, but it has other issues as I reported: the json generated have something different and then the json merger java application raises a null pointer exception.
Using alpha13 everything works, if vsyscall are emulated. 

> 
> Unfortunately, compilation of pplacer isn't trivial - or maybe it is my insufficient compilation skills. However, the latest version in bioconda is lacking an OSx package :-/

There is no need of the last bioconda version (see above).

@antgonza  to reproduce the issue, if you use Debian stretch 9.5 with kernel 4.18.0-0.bpo.1-amd64 you should face the problem. You should get the error reported in this ticket and inspecting /var/log/syslog you should see the vsyscall error, Or, more easily, you should get a segmentation fault by running pplacer --version
Instead, if my notes are correct, on Debian stretch 9.12 with kernel 4.9.0-12-amd64 you don't get the segmentation fault.

**Comment by antgonza** [comment]
@profgiuseppe, thank you for the info! 

[Looking at this](https://www.claudiokuenzler.com/blog/853/debian-wheezy-lxc-container-not-running-on-stretch-host-kernel-4.18-vsyscall), it seems like it "only" affects:
Debian 9 (Stretch) | 4.18.x.bpo | Yes!
Debian 10 (Buster) RC1 | 4.19.0 | Yes!
and if [I'm reading this correctly](https://en.wikipedia.org/wiki/Debian_version_history), it shouldn't affect `Bullseye` ... thus, perhaps updating to this version might help, not sure?

**Comment by sjanssen2** [comment]
@profgiuseppe I don't think we can ask the average user to compile their kernel with some non-default flags as our audience is mainly biologists.

To me, the missing OSx conda package of pplacer alpha19 is a show stopper and I don't have much experience with OSx myself. Are you able to build such a conda package for OSX?

I think I can tackle the json format error, but this is only a good investment of my time, if we can than in general, i.e. Linux and OSx, update to alpha19

**Comment by profgiuseppe** [comment]
Sorry @sjanssen2 , I hoped that sharing this information would shed a bit of light on the issue and help others in the same situation. Given that our client issue was solved with the kernel option, I cannot allocate time to build conda packages, also because we don't have any OSX system at hand to test it.

The kernel flag is an execution one, i.e. you should add it to grub and reboot the computer, no recompile needed. However I agree with you that this may be too much for the average biologist.
It is, however, something doable for IT managers of computing clusters.

I hope that you may find somebody that will help you with the OSX situation.
